tests/test_time_integrator.f90

```diff
@@ -185,13 +187,15 @@ program test_omp_adamsbashforth
     ! check order of convergence
     order = log(norm(nrun - 1)/norm(nrun))/log(2.0_dp)
     print *, 'order', order
-    if (abs(order - real(k, dp)) > 0.1_dp) then
+    if (abs(order - real(time_integrator%order, dp)) > 0.25_dp) then
       allpass = .false.
       write (stderr, '(a)') 'Check order... failed'
     else
       write (stderr, '(a)') 'Check order... passed'
     end if
 
+    call time_integrator%finalize
```

Comment:
Deallocate time_integrator memory for each scheme

Reply:
would this not be helpful as a code comment? after the PR is merged it will essentially "disappear" otherwise (and likewise for above comment).